### DISCUSSION TITLE
File IO on Main Thread is triggered whenever user types in Textfield using iPhone default Keyboard

### DISCUSSION THREAD

**Discussion by sumit11007** [getsentry/sentry-cocoa#3412] [Q&A]
Sentry is logging lot of Most slow frames when ever user is typing something in Textfield using iPhone default keyboard.
To Verify I have also added a default Textfield and when I try to type File IO is logged in Sentry.
"File IO on Main Thread
UITextSelectionInteraction._handleMultiTapGesture"
![image](https://github.com/getsentry/sentry-cocoa/assets/21600980/439f6a06-5129-4083-80a6-e0690ba91f6b)

**Comment by brustolin** [comment]
Thanks @sumit11007  for reporting this, we will investigate.

**Reply by philipphofmann** [reply]
@brustolin, we have an [ignore list](https://github.com/getsentry/sentry/blob/230daec21a79f805105371ae1eedd13bf300097f/src/sentry/utils/performance_issues/detectors/io_main_thread_detector.py#L109.) on the backend to ignore specific file extensions for file IO on the main thread. We may need to update that. We can't ignore all `.dat` files cause some apps might use it, maybe another file pattern could work.

**Reply by sumit11007** [reply]
Thanks @philipphofmann can we expect this in next release?

**Reply by brustolin** [reply]
Most likely any update in this regard will happen in the server side, so you dont need to update the Sentry SDK in your app.

**Reply by sumit11007** [reply]
Thanks @brustolin please let me know once its updated in server side.

**Reply by philipphofmann** [reply]
We created an issue for this problem https://github.com/getsentry/sentry-cocoa/issues/3442.

**Reply by andrejs-mamontovs** [reply]
Hello, I’m still receiving this message in version 8.55.1. Is there a recommended way to configure iOS to ignore it?

**Reply by philprime** [reply]
Hi @andrejs-mamontovs, try using a custom filter in `beforeSend` to drop any events you don't want. Otherwise please create a new issue so we can discuss it detail
